rj_gameplay/stp/role/assignment/naive.py

```diff
@@ -89,8 +99,24 @@ def compute_costs_matrix(
                 # Add to robot_costs.
                 robot_costs[robot_idx, request_idx] = cost
 
+            # Get cost of not assigning a robot to the request
+            unassigned_cost: float = request.cost_fn.unassigned_cost_fn(prev_result, world_state)
+
+            # Throw an exception if the returned cost is not finite.
+            if not isfinite(unassigned_cost):
```

Comment:
Just for my personal understanding, could you explain why we don't allow unassigneds to use infinite cost but we set invis robots to an infinite cost (lines 75-77 above)

Reply:
So that if a role is decidedly unassigned, it can't accidently be assigned due to it being finite/another robot also having infinite cost.